File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/authentication/AccountAuthenticatorService.java
package com.lattis.ellipse.authentication;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import androidx.annotation.Nullable;

public class AccountAuthenticatorService extends Service {

    AccountAuthenticator accountAuthenticator;

    @Override
    public void onCreate() {
       accountAuthenticator = new AccountAuthenticator(getApplicationContext());
    }

    @Nullable @Override
    public IBinder onBind(Intent intent) {
        return accountAuthenticator.getIBinder();
    }
}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/authentication/AccountAuthenticatorService.kt
package com.lattis.lattis.authentication

import android.accounts.AccountManager.ACTION_AUTHENTICATOR_INTENT
import android.app.Service
import android.content.Intent
import android.os.IBinder

class AccountAuthenticatorService : Service() {

    internal lateinit var accountAuthenticator: AccountAuthenticator

    override fun onCreate() {
        accountAuthenticator =
            AccountAuthenticator(applicationContext)
    }

    override fun onBind(intent: Intent): IBinder? {
        return if (intent.action == ACTION_AUTHENTICATOR_INTENT) getAuthenticator().iBinder else null
    }

    private fun getAuthenticator(): AccountAuthenticator {
        if (accountAuthenticator == null) {
            accountAuthenticator =
                AccountAuthenticator(applicationContext)
        }
        return accountAuthenticator
    }
}
